src/block-data-providers/bitcoin-core/provider.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { ConfigService } from '@nestjs/config';
+import { BitcoinCoreProvider } from '@/block-data-providers/bitcoin-core/provider';
+import {
+    bitcoinCoreConfig,
+    parsedTransactions,
+} from '@/block-data-providers/bitcoin-core/provider-fixtures';
+import { IndexerService } from '@/indexer/indexer.service';
+import { OperationStateService } from '@/operation-state/operation-state.service';
+import {
+    blockCountToHash,
+    blocks,
+    rawTransactions,
+} from '@/block-data-providers/bitcoin-core/provider-fixtures';
+describe('Bitcoincore Provider', () => {
+    let provider: BitcoinCoreProvider;
+
+    beforeEach(async () => {
+        const fakeConfigService = {
+            get: (key: string) => {
+                if (key == 'bitcoinCore') {
+                    return bitcoinCoreConfig;
+                }
+                if (key == 'app.network') {
+                    return 'regtest';
+                }
+                return null;
+            },
+        };
+        provider = new BitcoinCoreProvider(
+            fakeConfigService as unknown as ConfigService,
+            {} as unknown as IndexerService,
+            {} as unknown as OperationStateService,
+        );
```

Comment:
Please use the standard way of testing using `Test.createTestingModule`. See [this](https://github.com/Bitshala-Incubator/silent-pay-indexer/blob/main/src/indexer/indexer.service.spec.ts) for reference